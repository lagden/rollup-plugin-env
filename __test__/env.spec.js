'use strict'

import {rollup}from 'rollup'
import envs from '../src/env'

async function _code(bundle) {
	const {output: [{code}]} = await bundle.generate({
		format: 'es'
	})
	return code
}

test('options', async () => {
	const bundle = await rollup({
		input: '__fixtures__/in.js',
		plugins: [
			envs('test', {
				foo: 'bar'
			})
		]
	})

	const code = await _code(bundle)
	expect(code).toMatchSnapshot()
})

test('no options', async () => {
	const bundle = await rollup({
		input: '__fixtures__/in.js',
		plugins: [
			envs('test')
		]
	})

	const code = await _code(bundle)
	expect(code).toMatchSnapshot()
})
