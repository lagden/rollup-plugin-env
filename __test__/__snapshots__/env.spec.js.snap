// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no options 1`] = `
"globalThis.process = globalThis.process ?? {}
globalThis.process.env = globalThis.process.env ?? {}
globalThis.process.env['test'] = {}
for (const [k, v] of Object.entries({})) {
	globalThis.process.env['test'][k] = v
}


function foobar() {
	console.log('foobar');
}

foobar();
"
`;

exports[`options 1`] = `
"globalThis.process = globalThis.process ?? {}
globalThis.process.env = globalThis.process.env ?? {}
globalThis.process.env['test'] = {}
for (const [k, v] of Object.entries({\\"foo\\":\\"bar\\"})) {
	globalThis.process.env['test'][k] = v
}


function foobar() {
	console.log('foobar');
}

foobar();
"
`;
